HeterogeneousCore/AlpakaCore/interface/alpaka/ESProducer.h

```diff
@@ -69,12 +69,16 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
                          TReturn (T ::*iMethod)(device::Record<TRecord> const&),
                          edm::es::Label const& label = {}) {
       using TProduct = typename edm::eventsetup::produce::smart_pointer_traits<TReturn>::type;
-      if constexpr (detail::useESProductDirectly<TProduct>) {
+      if constexpr (detail::useESProductDirectly) {
         return Base::setWhatProduced(
             [iThis, iMethod](TRecord const& record) {
               auto const& devices = cms::alpakatools::devices<Platform>();
               assert(devices.size() == 1);
               device::Record<TRecord> const deviceRecord(record, devices.front());
+              static_assert(std::is_same_v<std::remove_cv_t<std::remove_reference_t<decltype(deviceRecord.queue())>>,
+                                           alpaka::Queue<Device, alpaka::Blocking>>,
+                            "Non-blocking queue when trying to use ES data product directly. This might indicate a "
+                            "need to extend the Alpaka ESProducer base class.");
```

Comment:
I could also just do
```suggestion
         alpaka::wait(deviceRecord.queue());    
```
instead, but I'd expect a CPU backend with a non-blocking Queue to have other consequences as well, so I thought a `static_assert` might be useful.